src/bootstrap/compile.rs

```diff
@@ -867,83 +872,84 @@ pub fn rustc_cargo_env(builder: &Builder<'_>, cargo: &mut Cargo, target: TargetS
         cargo.env("RUSTC_VERIFY_LLVM_IR", "1");
     }
 
-    // Pass down configuration from the LLVM build into the build of
-    // rustc_llvm and rustc_codegen_llvm.
-    //
     // Note that this is disabled if LLVM itself is disabled or we're in a check
     // build. If we are in a check build we still go ahead here presuming we've
     // detected that LLVM is already built and good to go which helps prevent
     // busting caches (e.g. like #71152).
-    if builder.config.llvm_enabled()
-        && (builder.kind != Kind::Check
-            || crate::llvm::prebuilt_llvm_config(builder, target).is_ok())
-    {
-        if builder.is_rust_llvm(target) {
-            cargo.env("LLVM_RUSTLLVM", "1");
-        }
-        let llvm::LlvmResult { llvm_config, .. } = builder.ensure(llvm::Llvm { target });
-        cargo.env("LLVM_CONFIG", &llvm_config);
-        if let Some(s) = target_config.and_then(|c| c.llvm_config.as_ref()) {
-            cargo.env("CFG_LLVM_ROOT", s);
-        }
+    let building_is_expensive = crate::llvm::prebuilt_llvm_config(builder, target).is_err();
+    // `top_stage == stage` might be false for `check --stage 1`, if we are building the stage 1 compiler
+    let can_skip_build = builder.kind == Kind::Check && builder.top_stage == stage;
+    let should_skip_build = building_is_expensive && can_skip_build;
+    if builder.config.llvm_enabled() && !should_skip_build {
+        rustc_llvm_env(builder, cargo, target)
+    }
```

Comment:
I would prefer running `prebuilt_llvm_config` only if llvm is enabled. Not a big deal, maybe it's just me:)

```suggestion
    if builder.config.llvm_enabled() {
        let building_is_expensive = crate::llvm::prebuilt_llvm_config(builder, target).is_err();
        // `top_stage == stage` might be false for `check --stage 1`, if we are building the stage 1 compiler
        let can_skip_build = builder.kind == Kind::Check && builder.top_stage == stage;
        let should_skip_build = building_is_expensive && can_skip_build;

        if !should_skip_build {
            rustc_llvm_env(builder, cargo, target)
        }
    }
```

Reply:
good idea, done :)